mongo/gridfs/bucket.go

```diff
@@ -455,6 +452,47 @@ func (b *Bucket) findChunks(ctx context.Context, fileID interface{}) (*mongo.Cur
 	return chunksCursor, nil
 }
 
+// returns true if the 2 index documents are equal
+func indexDocsEqual(expected, actual bsoncore.Document) (bool, error) {
+	if bytes.Equal(expected, actual) {
+		return true, nil
+	}
+
+	keyElemElems, err := actual.Elements()
+	if err != nil {
+		return false, err
+	}
+	modelKeysElems, err := expected.Elements()
+	if err != nil {
+		return false, err
+	}
+
+	if len(keyElemElems) != len(modelKeysElems) {
+		return false, nil
+	}
+
+	for _, elem := range keyElemElems {
+		key := elem.Key()
+		modelVal, err := expected.LookupErr(key)
+		if err != nil {
+			return false, nil
```

Comment:
I think this code will go away because the `LookupErr` call isn't necessary, but is the error intentionally being swallowed rather than returned?